readest: 0.9.38 -> 0.9.40; add TTS dependencies

ddogfoodd opened the pull request:
Changelog: https://github.com/readest/readest/releases/tag/v0.9.40
Diff: https://github.com/readest/readest/compare/v0.9.38...v0.9.40

add TTS dependencies, thanks to @eljamm !

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ddogfoodd committed:
readest: 0.9.38 -> 0.9.39

Changelog: https://github.com/readest/readest/releases/tag/v0.9.39
Diff: https://github.com/readest/readest/compare/v0.9.38...v0.9.39

eljamm commented:
> oh wait it's a pre-release

I think it was release at first, but then it changed to pre-release. Maybe we should investigate a bit further, but if this seems good then I personally see no issue in updating it.

eljamm commented:
So looking at the upstream issues, there appears to be some number of [regressions for Android](https://github.com/readest/readest/issues?q=is%3Aissue%20state%3Aopen%200.9.39), which might be the reason the release was changed to pre-release. This shouldn't really affect us and I indeed don't see anything wrong from testing this.

eljamm commented on pkgs/by-name/re/readest/package.nix:
```diff
@@ -19,13 +19,13 @@
+    hash = "sha256-1GVWQ5Kckh5od3p1hCfv8vIzCfCY8cRvCBmSZKcF+Qk=";
     fetchSubmodules = true;
   };
 
```
This has been bothering me for a while, but I finally figured out how to make TTS work by adding gstreamer and its plugins to the runtime deps:

```diff
   buildInputs = [
     webkitgtk_4_1
     gtk3
     librsvg
     openssl
+    # TTS
+    gst_all_1.gstreamer
+    gst_all_1.gstreamermm
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
   ];
```

ddogfoodd committed:
readest: add dependencies for TTS

thanks to @eljamm, see:
https://github.com/NixOS/nixpkgs/pull/404705#discussion_r2076233872

ddogfoodd committed:
readest: 0.9.39 -> 0.9.40

Changelog: https://github.com/readest/readest/releases/tag/v0.9.40
Diff: https://github.com/readest/readest/compare/v0.9.39...v0.9.40

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 404705`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/14901936505


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>readest</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>readest</li>
  </ul>
</details>

eljamm reviewed (approved):
0.9.40 is mostly a bugfix release of 0.9.39, the latter of which I've found no issues while testing. LGTM
